### PR TITLE
fix: UAT feedback and readmes

### PR DESCRIPTION
**Pull request by rutalreja-deloitte**
Fixes 

Readmes for 3,4,5
Missing provider.tf and backend.tf in 4-projects/shared
All Backend.tf in 4, 5 folder prefixes

### CODE REVIEW

File: 5-infrastructure/README.md [hunk 1]
```diff
@@ -6,44 +6,61 @@ All infrastructure components will be created using the base network created dur
...
 
-1. 0-bootstrap executed successfully.
-1. 1-org executed successfully.
-1. 2-environments executed successfully.
-1. 3-networks executed successfully.
-1. 4-projects executed successfully.
+1. [0-bootstrap](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/0-bootstrap/README.md) executed successfully.
+1. [1-org](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/1-org/README.md) executed successfully.
+1. [2-environments](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/2-environments/README.md) executed successfully.
+1. [3-networks](https://github.com/terraform-google-modules/terraform-example-foundation/blob/master/3-networks/README.md) executed successfully.
```
Comment (bharathkkb) [inline 5-infrastructure/README.md:31]: This should link to example-app 3-net

File: 5-infrastructure/README.md [hunk 2]
```diff
@@ -6,44 +6,61 @@ All infrastructure components will be created using the base network created dur
...
+1. [4-projects](../app-foundation/4-projects/README.md) executed successfully.
+
+**Troubleshooting:**
+If your user does not have access to run the commands above and you are in the organization admins group, you can append `--impersonate-service-account="boa-terraform-<z>-sa@prj-bu1-<z>-boa-sec-<xxxx>.iam.gserviceaccount.com"` for dev/npd/prd envs or `--impersonate-service-account="cicd-build-sa@prj-bu1-s-app-cicd-<xxxx>.iam.gserviceaccount.com"` for shared env to run the command as the terraform service account.
```
Comment (bharathkkb) [inline 5-infrastructure/README.md:35]: What are the comments above?

File: 5-infrastructure/business_unit_1/development/backend.tf [hunk 3]
```diff
@@ -17,6 +17,6 @@
 terraform {
   backend "gcs" {
     bucket = "UPDATE_ME"
-    prefix = "terraform/environments/development"
+    prefix = "terraform/infra/development"
```
Comment (bharathkkb) [inline 5-infrastructure/business_unit_1/development/backend.tf:20]:
```suggestion
    prefix = "terraform/app-infra/development"
```

File: 5-infrastructure/README.md (hunk 2 above)
Comment (rutalreja-deloitte) [reply 5-infrastructure/README.md]: changed to `**Troubleshooting:**
If your user does not have access to run the terraform modules locally and you are in the organization admins group, you can append `--impersonate-service-account=org-terraform@<SEED_PROJECT_ID>.iam.gserviceaccount.com` to run terraform modules as the service  account.`

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**